pettingzoo/test/parallel_test.py

```diff
@@ -114,7 +114,8 @@ def parallel_api_test(par_env: ParallelEnv, num_cycles=1000):
                     agent
                 ), "action_space should return the exact same space object (not a copy) for an agent (ensures that action space seeding works as expected). Consider decorating your action_space(self, agent) method with @functools.lru_cache(maxsize=None)"
 
-            for agent in live_agents:
+            # we make a copy of live_agents because live_agents changes during iteration
+            for agent in set(live_agents):
```

Comment:
Could we instead do something like:
```python
new_live_agents = []
for agent in live_agents:
  if terminated[agent] or truncated[agent]:
    break
  new_live_agents.append(agent)
live_agents = set(new_live_agents)
```

This would make it easier to read and for something like this to not happen in the future should someone else edit it.
Thanks!

Reply:
I've rewritten it to completely eliminate this problem. Let me know what you think.

Reply:
Awesome, thanks for the help!